### PR TITLE
Add vesting pallet

### LINKED ISSUE #376: [Runtime] Add the Vesting pallet from Substrate FRAME
We will need to add the Vesting pallet for Substrate FRAME in order to handle vesting logic.

### PR DESCRIPTION
**Pull request by sea212**
depends on #373 
closes #376

### CODE REVIEW

**Comment by sea212** [comment]
Blocked until #373 is merged

**Review by lsaether — APPROVED** [review]

**Pull request merged**
